--- drivers/spi/Kconfig	2026-03-03 13:25:59.925572489 +0000
+++ /home/jonathan/sonata-linux-clone/linux-xip/drivers/spi/Kconfig	2026-03-07 19:02:43.157411845 +0000
@@ -93,6 +93,11 @@
 	  This enables support for the SPI controller present on the
 	  Qualcomm Atheros AR934X/QCA95XX SoCs.
 
+config SPI_OPENTITAN
+	tristate "OpenTitan SPI controller"
+	help
+	  SPI controller for OpenTitan-derived SoCs (lowRISC Sonata).
+
 config SPI_ATH79
 	tristate "Atheros AR71XX/AR724X/AR913X SPI controller driver"
 	depends on ATH79 || COMPILE_TEST
--- drivers/spi/Makefile	2026-03-03 13:25:59.925664236 +0000
+++ /home/jonathan/sonata-linux-clone/linux-xip/drivers/spi/Makefile	2026-03-07 19:02:43.157411845 +0000
@@ -158,3 +158,4 @@
 # SPI slave protocol handlers
 obj-$(CONFIG_SPI_SLAVE_TIME)		+= spi-slave-time.o
 obj-$(CONFIG_SPI_SLAVE_SYSTEM_CONTROL)	+= spi-slave-system-control.o
+obj-$(CONFIG_SPI_OPENTITAN)		+= spi-opentitan.o
