--- a/drivers/spi/Kconfig
+++ b/drivers/spi/Kconfig
@@ -93,6 +93,11 @@
 	  This enables support for the SPI controller present on the
 	  Qualcomm Atheros AR934X/QCA95XX SoCs.

+config SPI_OPENTITAN
+	tristate "OpenTitan SPI controller"
+	help
+	  SPI controller for OpenTitan-derived SoCs (lowRISC Sonata).
+
 config SPI_ATH79
 	tristate "Atheros AR71XX/AR724X/AR913X SPI controller driver"
 	depends on ATH79 || COMPILE_TEST
--- a/drivers/spi/Makefile
+++ b/drivers/spi/Makefile
@@ -158,3 +158,4 @@
 # SPI slave protocol handlers
 obj-$(CONFIG_SPI_SLAVE_TIME)		+= spi-slave-time.o
 obj-$(CONFIG_SPI_SLAVE_SYSTEM_CONTROL)	+= spi-slave-system-control.o
+obj-$(CONFIG_SPI_OPENTITAN)		+= spi-opentitan.o
